### ISSUE TITLE
Docker module fails to parse envvars that are empty strings

### ISSUE THREAD

**Issue by mattupstate** [ansible/ansible-modules-core#4466]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

docker module
##### ANSIBLE VERSION

```
$ .venv/bin/ansible --version
ansible 2.1.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

Ubuntu Xenial
##### SUMMARY

Sometimes I'm setting the value of an environment variable for a Docker container as an empty string. When the container is started for the first time all is well. However, when the container is already running, and the `docker` module compares the existing state, it fails when parsing the environment variables.
##### STEPS TO REPRODUCE

Run a playbook with a `docker` module task with an empty envvar twice against the same host.

```
- name: "run"
  docker:
    image: nginx
    name: nginx
    net: host
    env:
      SOMETHING_NOT_EMPTY: hello
      SOMETHING_EMPTY: 
    state: reloaded
```
##### EXPECTED RESULTS

The module to not fail
##### ACTUAL RESULTS

On the second run:

```
fatal: [machine]: FAILED! => {"changed": false, "failed": true, "module_stderr": "", "module_stdout": "\r\nTraceback (most recent call last):\r\n  File \"/home/vagrant/.ansible/tmp/ansible-tmp-1471550804.18-264229313132160/docker\", line 3752, in <module>\r\n    main()\r\n  File \"/home/vagrant/.ansible/tmp/ansible-tmp-1471550804.18-264229313132160/docker\", line 1740, in main\r\n    reloaded(manager, containers, count, name)\r\n  File \"/home/vagrant/.ansible/tmp/ansible-tmp-1471550804.18-264229313132160/docker\", line 1599, in reloaded\r\n    for container in manager.get_differing_containers():\r\n  File \"/home/vagrant/.ansible/tmp/ansible-tmp-1471550804.18-264229313132160/docker\", line 1136, in get_differing_containers\r\n    name, value = container_env.split('=', 1)\r\nValueError: need more than 1 value to unpack\r\n", "msg": "MODULE FAILURE", "parsed": false}
```

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by zfil** [comment]
wontfix
this module is deprecated
@cove, @softzilla, @zfil, @ThomasSteinbach guys, in general I suggest we stop docker modules needing docker-py to rely only on native docker component (like compose)
